src/Resources/BaseDbTableResource.php

```diff
@@ -2246,6 +2246,10 @@ protected function retrieveRelationRecords(TableSchema $schema, RelationSchema $
                             $extras['fields'] = $fields;
                         }
                         $relatedRecords = $this->retrieveVirtualRecords($refService, '_table/' . $refTable, $extras);
+                        // For some bizarre reason, on Debian 9, duplicates of related tables are returned. (It returns the related table, multiplied by the 
+                        // total number of records, e.g 8 records, 8 duplicates.) In fear of breaking things elsewhere, the below will check and delete
+                        // any duplicate entries and update. If there aren't any to begin with, then nothing will change.
+                        $relatedRecords = array_unique($relatedRecords, SORT_REGULAR);
```

Comment:
Although the line looks innocent, it raised a couple of questions in my head. 
1. What about other retrieve*() methods? 
We have retrieveRecords, retrieveRecordsByIds, retrieveRelatedRecords, retrieveRelationRecords, retrieveRecordsByFilter. Are they also affected?
I know it is a lot to test and I think for most of them the problem would not reproduce that is why I think we have to get to the bottom of the issue. Not to fix it with `array_unique` but to figure out why those duplicates are there. Maybe they just appear because of some database configs and we don't need to change the core code.
2. The default limit for the number of records is 1000 but I can remove that limit and my response would be 10000 records. Going through all of them with `array_unique` could make things slow. In PHP7.x they fixed the slowness of the `array_unique` but it is still an O(n).
Here is a couple of links on the topic: [stackoverflow](https://stackoverflow.com/questions/8321620/array-unique-vs-array-flip), [2021 article](https://tideways.com/profiler/blog/innocent-looking-array_unique-2-stories-of-performance-hogs-in-shopware-6-and-tideways-own-backend-code-base)